robottelo/utils/sso.py

```diff
@@ -0,0 +1,197 @@
+        self.execute(f"{KEY_CLOAK_CLI} delete -r {settings.rhsso.realm} users/{user_details['id']}")
+
+    def create_group(self, group_name=None):
+        """Create the RHSSO group"""
+        if not group_name:
+            group_name = gen_string('alphanumeric')
+        RHSSO_NEW_GROUP['name'] = group_name
+        self.upload_rhsso_entity(RHSSO_NEW_GROUP, "create_group")
+        result = self.execute(
+            f"{KEY_CLOAK_CLI} create groups -r {settings.rhsso.realm} -f create_group"
+        )
+        return result
+
+    def delete_rhsso_group(self, group_name):
+        """Delete the RHSSO group"""
+        group_details = self.get_rhsso_groups_details(group_name)
+        self.execute(
+            f"{KEY_CLOAK_CLI} delete -r {settings.rhsso.realm} groups/{group_details['id']}"
+        )
+
+    def update_client_configuration(self, json_content):
+        """Update the client configuration"""
+        client_id = self.get_rhsso_client_id()
+        self.upload_rhsso_entity(json_content, "update_client_info")
+        update_cmd = (
+            f"{KEY_CLOAK_CLI} update clients/{client_id}"
+            f"-f update_client_info -s enabled=true --merge"
+        )
+        self.execute(update_cmd)
+
+    def get_oidc_token_endpoint(self):
+        """getter oidc token endpoint"""
+        return (
+            f"https://{settings.rhsso.host_name}/auth/realms/"
+            f"{settings.rhsso.realm}/protocol/openid-connect/token"
+        )
+
+    def get_oidc_client_id(self):
+        """getter for the oidc client_id"""
+        return f"{settings.server.hostname}-foreman-openidc"
+
+    def get_oidc_authorization_endpoint(self):
+        """getter for the oidc authorization endpoint"""
+        return (
+            f"https://{settings.rhsso.host_name}/auth/realms/"
+            f"{settings.rhsso.realm}/protocol/openid-connect/auth"
+        )
+
+    def get_two_factor_token_rh_sso_url(self):
+        """getter for the two factor token rh_sso url"""
+        return (
+            f"https://{settings.rhsso.host_name}/auth/realms/"
+            f"{settings.rhsso.realm}/protocol/openid-connect/"
+            f"auth?response_type=code&client_id={settings.server.hostname}-foreman-openidc&"
+            "redirect_uri=urn:ietf:wg:oauth:2.0:oob&scope=openid"
+        )
+
+    @contextmanager
+    def open_pxssh_session(
+        ssh_key=settings.server.ssh_key,
+        hostname=settings.server.get('hostname', None),
+        username=settings.server.ssh_username,
+    ):
+        ssh_options = {'IdentityAgent': ssh_key}
+        ssh_session = pxssh.pxssh(options=ssh_options)
+        ssh_session.login(hostname, username, sync_multiplier=5)
+        yield ssh_session
+        ssh_session.logout()
+
+    def set_the_redirect_uri(self):
+        client_config = {
+            "redirectUris": [
+                "urn:ietf:wg:oauth:2.0:oob",
+                f"https://{settings.server.hostname}/users/extlogin/redirect_uri",
```

Comment:
This should reference `self.satellite` - but requires basing the class off `ContentHost` to have access.

otherwise, any functions dealing with Satellite need to take a required `sat` kwarg.